Use GoogleFileStore in GooglePhotosExporter to support large accounts.

Currently if a user has a lot of photo albums or photos, we hit this error when the job is processing - we can do better things to fix this at the core architecture level rather than each extension but for now we should fix the PhotoExporter to just use filestore 


com.google.cloud.datastore.DatastoreException: The value of property "org.datatransferproject.spi.transfer.types.TempPhotosData" is longer than 1500 bytes. at com.google.cloud.datastore.spi.v1.HttpDatastoreRpc.translate(HttpDatastoreRpc.java:129) at com.google.cloud.datastore.spi.v1.HttpDatastoreRpc.commit(HttpDatastoreRpc.java:155) at com.google.cloud.datastore.DatastoreImpl$5.call(DatastoreImpl.java:418) at com.google.cloud.datastore.DatastoreImpl$5.call(DatastoreImpl.java:415) at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:89) at com.google.cloud.RetryHelper.run(RetryHelper.java:74) at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:51) at com.google.cloud.datastore.DatastoreImpl.commit(DatastoreImpl.java:414) at com.google.cloud.datastore.TransactionImpl.commit(TransactionImpl.java:100) at org.datatransferproject.cloud.google.GoogleJobStore.create(GoogleJobStore.java:263) at org.datatransferproject.datatransfer.google.photos.GooglePhotosExporter.populateContainedPhotosList(GooglePhotosExporter.java:239) at org.datatransferproject.datatransfer.google.photos.GooglePhotosExporter.export(GooglePhotosExporter.java:83) at org.datatransferproject.datatransfer.google.photos.GooglePhotosExporter.export(GooglePhotosExporter.java:51) at org.datatransferproject.transfer.CallableExporter.call(CallableExporter.java:50) at org.datatransferproject.transfer.CallableExporter.call(CallableExporter.java:31) at org.datatransferproject.types.transfer.retry.RetryingCallable.call(RetryingCallable.java:64) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:110) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copy(PortabilityInMemoryDataCopier.java:75) at org.datatransferproject.transfer.JobProcessor.processJob(JobProcessor.java:98) at org.datatransferproject.transfer.Worker.doWork(Worker.java:33) at org.datatransferproject.transfer.WorkerMain.poll(WorkerMain.java:164) at org.datatransferproject.transfer.WorkerMain.main(WorkerMain.java:67) Caused by: com.google.datastore.v1.client.DatastoreException: The value of property "org.datatransferproject.spi.transfer.types.TempPhotosData" is longer than 1500 bytes., code=INVALID_ARGUMENT at com.google.datastore.v1.client.RemoteRpc.makeException(RemoteRpc.java:226) at com.google.datastore.v1.client.RemoteRpc.makeException(RemoteRpc.java:275) at com.google.datastore.v1.client.RemoteRpc.call(RemoteRpc.java:186) at com.google.datastore.v1.client.Datastore.commit(Datastore.java:87) at com.google.cloud.datastore.spi.v1.HttpDatastoreRpc.commit(HttpDatastoreRpc.java:153) ... 20 more

--- a/extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java
+++ b/extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java
@@ -254,8 +254,6 @@ public class GooglePhotosExporter
     // should consider putting logic in JobStore itself to handle it
     InputStream stream = convertJsonToInputStream(tempPhotosData);
     jobStore.create(jobId, createCacheKey(), stream);
-
-    jobStore.create(jobId, createCacheKey(), tempPhotosData);
   }
 
   @VisibleForTesting

--- a/extensions/data-transfer/portability-data-transfer-google/src/test/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporterTest.java
+++ b/extensions/data-transfer/portability-data-transfer-google/src/test/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporterTest.java
@@ -24,6 +24,7 @@ import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyZeroInteractions;
 import static org.mockito.Mockito.when;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.api.client.json.jackson2.JacksonFactory;
 import java.io.IOException;
 import java.io.InputStream;
@@ -247,11 +248,13 @@ public class GooglePhotosExporterTest {
     googlePhotosExporter.populateContainedPhotosList(uuid, null);
 
     // Check contents of job store
-    ArgumentCaptor<TempPhotosData> tempPhotosDataArgumentCaptor = ArgumentCaptor
-        .forClass(TempPhotosData.class);
-    verify(jobStore).create(Matchers.eq(uuid), Matchers.eq("tempPhotosData"), tempPhotosDataArgumentCaptor.capture());
-    assertThat(tempPhotosDataArgumentCaptor.getValue().lookupContainedPhotoIds())
-        .containsExactly(PHOTO_ID, secondId);
+    ArgumentCaptor<InputStream> inputStreamArgumentCaptor = ArgumentCaptor
+        .forClass(InputStream.class);
+    verify(jobStore).create(Matchers.eq(uuid), Matchers.eq("tempPhotosData"),
+        inputStreamArgumentCaptor.capture());
+    TempPhotosData tempPhotosData = new ObjectMapper()
+        .readValue(inputStreamArgumentCaptor.getValue(), TempPhotosData.class);
+    assertThat(tempPhotosData.lookupContainedPhotoIds()).containsExactly(PHOTO_ID, secondId);
   }
 
   @Test

